pkg/clusterconditions/promql/promql.go

```diff
@@ -98,6 +101,31 @@ func (p *PromQL) Valid(ctx context.Context, condition *configv1.ClusterCondition
 	return nil
 }
 
+// Type CustomClient encapsulates an API client interface.
+type CustomClient struct {
+	client api.Client
+}
+
+// Returns a response containing status code 501 for POST requests. All other requests are
+// propagated into the encapsulated API client interface inside the CustomClient object.
+// This is required due to the thanos-querier.openshift-monitoring.svc tenancy port.
+func (c *CustomClient) Do(ctx context.Context, req *http.Request) (*http.Response, []byte, error) {
+	if req.Method == http.MethodPost {
+		if req.Body != nil {
+			req.Body.Close() // Underlying RoundTripper is normally responsible to close the body
+		}
+		return &http.Response{
+			StatusCode: http.StatusNotImplemented,
+			Body:       io.NopCloser(strings.NewReader("")), // http.Response.Body must not nill as per documentation
+		}, nil, nil
```

Comment:
Oh, maybe I should return an empty `[]byte` instead of a `nil`. I will address this tomorrow.

Reply:
`nil` should be fine, because consumers should be coded defensively to handle the `nil` case.  If you suspect they aren't, making those consumers more robust is probably better value than moving to `[]byte` here.  But if the consumers are outside our control, moving to `[]byte` here would be a convenient workaround.

Reply:
The downstream client code for creating the `body []byte`:

https://github.com/openshift/cluster-version-operator/blob/101af608a43bf8ae6228893db3f86de6346d0ecb/vendor/github.com/prometheus/client_golang/api/client.go#L136-L143

[Seems to return an empty slice for an empty body](https://go.dev/play/p/Lm7acVaUJGc), so I'll modify the code to return `[]byte{}` here instead of `nil`. 

The consumer of this returned `body` is the Prometheus library code:

https://github.com/openshift/cluster-version-operator/blob/80e7bbda9dadd86c067e09458a9a828d0cc3a1c2/vendor/github.com/prometheus/client_golang/api/prometheus/v1/api.go#L1382

It seems to be robust. Since the HTTP response body must not be `nil`, it would make sense that the `body []byte` variable would also be an empty slice for an empty body.